core/tests/DesktopTests/AdalHeadlessAutomationTestApp/App/AuthenticationHelper.cs

```diff
@@ -0,0 +1,375 @@
+﻿using Microsoft.IdentityModel.Clients.ActiveDirectory;
+using Newtonsoft.Json;
+using System;
+using System.Collections.Generic;
+using System.IO;
+using System.Linq;
+using System.Runtime.Serialization.Json;
+using System.Threading.Tasks;
+using System.Web.Script.Serialization;
+using Microsoft.IdentityModel.Clients.ActiveDirectory.Internal;
+using Microsoft.IdentityModel.Clients.ActiveDirectory.Internal.Cache;
+
+// COPIED FROM ADAL.NET repo test automation app
+// https://github.com/AzureAD/azure-activedirectory-library-for-dotnet/blob/dev/automation/WinFormsAutomationApp/AuthenticationHelper.cs
+
+namespace WinFormsAutomationApp
+{
+    internal static class AuthenticationHelper
+    {
+        #region Public Methods      
+        public static async Task<string> AcquireToken(Dictionary<string, string> input)
```

Comment:
You are bypassing C# strong typing by using a dictionary as "input". This has a number of issues - you can't really tell what a method needs to run, you now have to maintain that dictionary (do I have to delete "password" and add "user_identifier_type" ?

I understand that it would take some time to refactor this, but I think we should take that time to improve the long term maintenance of our solution.